src/main/java/com/palantir/gradle/versions/VersionsLockPlugin.java

```diff
@@ -527,6 +519,7 @@ private void recursivelyCopyProjectDependenciesWithScope(
             Project currentProject,
             DependencySet dependencySet,
             Map<Configuration, String> copiedConfigurationsCache,
+            Map<ModuleIdentifier, GcvScope> scopes,
```

Comment:
Can we wrap this Map into it's own type? Then we can put low level the behaviour of setting a value with the comparator/getting it out again etc into that object.

Reply:
yep makes sense, done